provisioning/service/src/IndividualEnrollmentsClient.cs

```diff
@@ -0,0 +1,258 @@
+        /// </summary>
+        /// <param name="individualEnrollment">The individual enrollment.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <exception cref="ArgumentNullException">If the provided <paramref name="individualEnrollment"/> is null.</exception>
+        /// <exception cref="DeviceProvisioningServiceException">
+        /// If the client failed to send the request or service was not able to execute the operation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided <paramref name="cancellationToken"/> has requested cancellation.</exception>
+        public async Task DeleteAsync(IndividualEnrollment individualEnrollment, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(individualEnrollment, nameof(individualEnrollment));
+
+            await _contractApiHttp
+                .RequestAsync(
+                    HttpMethod.Delete,
+                    GetEnrollmentUri(individualEnrollment.RegistrationId),
+                    null,
+                    null,
+                    individualEnrollment.ETag,
+                    cancellationToken)
+                .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Create, update or delete a set of individual Device Enrollments.
+        /// </summary>
+        /// <remarks>
+        /// This API provide the means to do a single operation over multiple individualEnrollments. A valid operation
+        /// is determined by <see cref="BulkOperationMode"/>, and can be 'create', 'update', 'updateIfMatchETag', or 'delete'.
+        /// </remarks>
+        /// <param name="bulkOperationMode">The <see cref="BulkOperationMode"/> that defines the single operation to do over the individualEnrollments. It cannot be null.</param>
+        /// <param name="individualEnrollments">The collection of <see cref="IndividualEnrollment"/> that contains the description of each individualEnrollment. It cannot be null or empty.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A <see cref="BulkEnrollmentOperationResult"/> object with the result of operation for each enrollment.</returns>
+        /// <exception cref="ArgumentNullException">If the provided <paramref name="individualEnrollments"/> is null.</exception>
+        /// <exception cref="ArgumentException">If the provided <paramref name="individualEnrollments"/> is an empty collection.</exception>
+        /// <exception cref="DeviceProvisioningServiceException">
+        /// If the client failed to send the request or service was not able to execute the bulk operation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided <paramref name="cancellationToken"/> has requested cancellation.</exception>
+        public async Task<BulkEnrollmentOperationResult> RunBulkEnrollmentOperationAsync(
```

Comment:
should we rename the method to 'RunBulkOperationAsync'?

Reply:
I like it, sure. "Enrollment" is implied by the client, so it makes sense to me.